File: sorting_algorithms/first_occulotion/first_occulotion.py
def first_occulotion(array: list, target: int):
    low = 0
    high = len(array) - 1
    
    while(low < high):
        mid = low + (high - low) // 2
        
        if len(array) == 0:
            print(" xxx array is empty xxx ")
            break
        if (array[mid] < target):
            low = mid + 1
        else:
            high = mid
    
    if (array[low] == target):
        return low
    
array = list(map(int, input("Enter elemnts array: ").split(",")))
target = int(input("Enter target: "))

print(first_occulotion(array, target))

File: sorting_algorithms/first_occulotion/first_occulotion.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

vector<int> get_value(){
    std::string input_str;
    vector<int> element;
    
    std::cout << "Enter elemnts array(separate elements with commas): ";
    getline(std::cin, input_str);
    stringstream ss(input_str);
    std::string temp;
    char delimiter = ',';
    while (getline(ss,temp, delimiter))
    {
        element.push_back(std::stoi(temp));
    }

    return element;
}

int main(){
    vector<int> array = get_value();
    int low = 0;
    int high = array.size() - 1;
    int target;
    
    cout << "Enter target: ";
    cin >> target;

    if(array.empty()){
        cerr << "xxx array is empty xxx" << endl;
        return 0; 
    }

    while (low < high)
    {
        int mid = low + (high - low) / 2;

        if(target > array[mid])
            low = mid + 1;
        else
            high = mid;

    }
    if(target == array[low]){
        cout << "find target, index is: " << low << endl;
    }
    else{
        cout << "Not found target !!! " << endl;
    }
    return 0;
}
